cirq-core/cirq/conftest.py

```diff
@@ -1,6 +1,12 @@
 import inspect
-
+import os
 import matplotlib.pyplot as plt
+import pytest
+
+
+@pytest.fixture(scope="session", autouse=True)
+def setup_cirq_testing_envvar():
+    os.environ['CIRQ_TESTING'] = "true"
```

Comment:
Can we put this line in `pytest_configure` instead? Also, will this get picked up if running tests on other modules such as `cirq-google`?

Reply:
It does get picked up by cirq-google.
I have to check if `pytest_configure` works or not - I wanted the setting before the tests execute but _after_ the test collection, to allow for collection code to use deprecated modules - for example "module crawlers" like the json_serialization_test.py list all public objects, and during that access `cirq.google`, which triggers the error mechanism. Maybe I should just turn it off during collection for those specific cases?

Reply:
I think using `assert_deprecated` around the module discovery would make sence, since it explicitly documents where we are using deprecated behavior. That seems preferable to relying on specific execution order of pytest_configure versus a session fixture.

Reply:
Will do that!